### ISSUE TITLE
Global parameters together with confirm_changeset results in AttributeError 

### ISSUE THREAD

**Issue by lobbin** [aws/aws-sam-cli#2923]
### Description:

Adding a global parameters section, such as "[default.global.parameters]" to samconfig.toml makes the sam-cli break with AttributeError if confirm_changeset is present in deploy configuration.

### Steps to reproduce:

sam init, choice: 1, 1, 7, 1

Provide default samconfig.toml with the following content
```
version = 0.1

[default.global.parameters]

[default.deploy.parameters]
stack_name = "Test"
confirm_changeset = true
```

### Observed result:

```
Traceback (most recent call last):
  File "/usr/local/bin/sam", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1257, in invoke
    sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 700, in make_context
    self.parse_args(ctx, args)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1048, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1623, in handle_parse_result
    value = self.full_process_value(ctx, value)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1965, in full_process_value
    return Parameter.full_process_value(self, ctx, value)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1589, in full_process_value
    value = self.process_value(ctx, value)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1579, in process_value
    return self.type_cast_value(ctx, value)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1568, in type_cast_value
    return _convert(value, (self.nargs != 1) + bool(self.multiple))
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/core.py", line 1565, in _convert
    return self.type(value, self, ctx)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/types.py", line 46, in __call__
    return self.convert(value, param, ctx)
  File "/usr/local/Cellar/aws-sam-cli/1.23.0/libexec/lib/python3.8/site-packages/click/types.py", line 412, in convert
    value = value.lower()
AttributeError: 'Bool' object has no attribute 'lower'
```

### Expected result:

Working deployment.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS Big Sur 11.3.1
2. `sam --version`: SAM CLI, version 1.23.0
3. AWS region: N/A

`Add --debug flag to command you are running`

**Comment by khiami** [comment]
I faced the same after sam cli updated the `samconfig.toml` file to include `confirm_changeset = true`

**Comment by wchengru** [comment]
Hi @khiami , the fix on this issue is released in SAM CLI 1.25.0. Can you please upgrade SAM CLI to 1.25.0 and try again?

I will close this one. Please feel free to reopen it if it doesn't fix your issue.
